core/src/test/java/org/apache/camel/kafkaconnector/CamelSourceTaskTest.java

```diff
@@ -32,270 +31,239 @@
 
-        assertEquals(2, camelSourceTask.getCms().getEndpoints().size());
+        List<SourceRecord> poll3 = sourceTask.poll();
+        assertEquals(1, poll3.size());
+        assertNull(poll3.get(0).key());
+        assertNull(poll3.get(0).keySchema());
 
-        camelSourceTask.getCms().getEndpoints().stream()
-                .filter(e -> e.getEndpointUri().startsWith("direct"))
-                .forEach(e -> {
-                    assertTrue(e.getEndpointUri().contains("end"));
-                    assertTrue(e.getEndpointUri().contains("pollingConsumerBlockTimeout=1000"));
-                    assertTrue(e.getEndpointUri().contains("pollingConsumerBlockWhenFull=false"));
-                    assertTrue(e.getEndpointUri().contains("pollingConsumerQueueSize=10"));
-                });
+        sourceTask.stop();
+    }
 
-        camelSourceTask.stop();
+    @Test
+    public void testSourcePollingWithBody() {
+        Map<String, String> props = new HashMap<>();
+        props.put(CamelSourceConnectorConfig.TOPIC_CONF, TOPIC_NAME);
+        props.put(CamelSourceConnectorConfig.CAMEL_SOURCE_URL_CONF, DIRECT_URI);
+
+        CamelSourceTask sourceTask = new CamelSourceTask();
+        sourceTask.start(props);
+        final ProducerTemplate template = sourceTask.getCms().createProducerTemplate();
+
+        // send String
+        template.sendBody(DIRECT_URI, "test");
+
+        List<SourceRecord> poll1 = sourceTask.poll();
+        assertEquals(1, poll1.size());
+        assertEquals("test", poll1.get(0).value());
+        assertEquals(Schema.Type.STRING, poll1.get(0).valueSchema().type());
+        assertNull(poll1.get(0).key());
+        assertNull(poll1.get(0).keySchema());
+
+        // send boolean
+        template.sendBody(DIRECT_URI, true);
+
+        List<SourceRecord> poll2 = sourceTask.poll();
+        assertEquals(1, poll2.size());
+        assertTrue((boolean)poll2.get(0).value());
+        assertEquals(Schema.Type.BOOLEAN, poll2.get(0).valueSchema().type());
+        assertNull(poll2.get(0).key());
+        assertNull(poll2.get(0).keySchema());
+
+        // send long
+        template.sendBody(DIRECT_URI, 1234L);
+
+        List<SourceRecord> poll3 = sourceTask.poll();
+        assertEquals(1, poll3.size());
+        assertEquals(1234L, poll3.get(0).value());
+        assertEquals(Schema.Type.INT64, poll3.get(0).valueSchema().type());
+        assertNull(poll3.get(0).key());
+        assertNull(poll3.get(0).keySchema());
+
+        // send null
+        template.sendBody(DIRECT_URI, null);
+
+        List<SourceRecord> poll4 = sourceTask.poll();
+        assertNull(poll4.get(0).key());
+        assertNull(poll4.get(0).keySchema());
+        assertNull(poll4.get(0).value());
+        assertNull(poll4.get(0).valueSchema());
+
+        sourceTask.stop();
     }
 
     @Test
-    public void testSourceUrlPrecedenceOnComponentProperty() {
+    public void testUrlPrecedenceOnComponentProperty() {
```

Comment:
I think this version is not testing what the old version was testing since `endpointProperty` and `pathChunk` are not legal properties of `direct` component.

Reply:
Fixed, thanks.

Reply:
Thanks!